bundles/org.openhab.binding.linky/src/main/java/org/openhab/binding/linky/internal/LinkyAuthService.java

```diff
@@ -0,0 +1,157 @@
+                    httpService.createDefaultHttpContext());
+            httpService.registerResources(LinkyBindingConstants.LINKY_ALIAS + LinkyBindingConstants.LINKY_IMG_ALIAS,
+                    "web", null);
+        } catch (NamespaceException | ServletException | IOException e) {
+            logger.warn("Error during linky servlet startup", e);
+        }
+    }
+
+    @Deactivate
+    protected void deactivate(ComponentContext componentContext) {
+        httpService.unregister(LinkyBindingConstants.LINKY_ALIAS);
+        httpService.unregister(LinkyBindingConstants.LINKY_ALIAS + LinkyBindingConstants.LINKY_IMG_ALIAS);
+    }
+
+    /**
+     * Creates a new {@link LinkyAuthServlet}.
+     *
+     * @return the newly created servlet
+     * @throws IOException thrown when an HTML template could not be read
+     */
+    private HttpServlet createServlet() throws IOException {
+        return new LinkyAuthServlet(this, readTemplate(TEMPLATE_INDEX));
+    }
+
+    /**
+     * Reads a template from file and returns the content as String.
+     *
+     * @param templateName name of the template file to read
+     * @return The content of the template file
+     * @throws IOException thrown when an HTML template could not be read
+     */
+    private String readTemplate(String templateName) throws IOException {
+        final URL index = bundleContext.getBundle().getEntry(templateName);
+
+        if (index == null) {
+            throw new FileNotFoundException(
+                    String.format("Cannot find '{}' - failed to initialize Linky servlet", templateName));
+        } else {
+            try (InputStream inputStream = index.openStream()) {
+                return new String(inputStream.readAllBytes(), StandardCharsets.UTF_8);
+            }
+        }
+    }
+
+    /**
+     * Call with Linky redirect uri returned State and Code values to get the refresh and access tokens and persist
+     * these values
+     *
+     * @param servletBaseURL the servlet base, which will be the Linky redirect url
+     * @param state The Linky returned state value
+     * @param code The Linky returned code value
+     * @return returns the name of the Linky user that is authorized
+     */
+    public String authorize(String servletBaseURL, String state, String code) {
+        LinkyAccountHandler accountHandler = getLinkyAccountHandler();
+        if (accountHandler == null) {
+            logger.debug(
+                    "Linky redirected with state '{}' but no matching bridge was found. Possible bridge has been removed.",
+                    state);
+            throw new RuntimeException(ERROR_UKNOWN_BRIDGE);
+        } else {
+            return accountHandler.authorize(servletBaseURL, code);
+        }
+    }
+
+    /**
+     * @param listener Adds the given handler
+     */
+    public void setLinkyAccountHandler(LinkyAccountHandler accountHandler) {
```

Comment:
`LinkyAccountHandler` could be passed by `@Reference` in the constructor ?

Reply:
this class was remove from the new version after refactoring the code !